tests/test_doc_examples.py

```diff
@@ -53,34 +85,23 @@ class DocExampleTest(BaseTest):
           sys.version_info.major == 2 and
           sys.version_info.minor == 7)))
 
+    providers = [[c] for c in clouds.keys()]
+
     @skipif(skip_condition, reason="Doc tests must be explicitly enabled with C7N_DOC_TEST")
-    def test_doc_examples(self):
-        policies = []
-        policy_map = {}
-        idx = 1
-        for ptext, resource_name, el_name in get_doc_examples():
-            data = yaml.safe_load(ptext)
-            for p in data.get('policies', []):
-                # We unique based on name and content to avoid duplicates
-                # from inherited docs.
-                if p['name'] in policy_map:
-                    if p != policy_map[p['name']]:
-                        # Give each policy a unique name with enough
-                        # context that we can identify the origin on
-                        # failures.
-                        p['name'] = "%s-%s-%s-%d" % (
-                            resource_name.split('.')[-1],
-                            el_name,
-                            p.get('name', 'unknown'), idx)
-                    continue
-                policy_map[p['name']] = p
+    @parameterized.expand(providers)
+    def test_doc_examples(self, provider):
+        policies, duplicate_names = get_doc_policies(resources(cloud_provider=provider))
+        self.load_policy_set({'policies': [v for v in policies.values()]})
+
+        # TODO: This check needs to be enabled when duplicate policy names are cleaned up
+        if provider != 'aws':
+            self.assertSetEqual(duplicate_names, set())
+
+        if provider == 'aws':
+            for p in policies.values():
                 # Note max name size here is 54 if its a lambda policy
                 # given our default prefix custodian- to stay under 64
                 # char limit on lambda function names.
                 if len(p['name']) >= 54 and 'mode' in p:
                     raise ValueError(
-                        "doc policy exceeds name limit resource:%s element:%s policy:%s" % (
-                            resource_name, el_name, p['name']))
-                policies.append(p)
-                idx += 1
-        self.load_policy_set({'policies': policies})
+                        "doc policy exceeds name limit policy:%s" % (p['name']))
```

Comment:
fwiw the limit on 64 is common to gcp and aws function names.. what's the limit for azure?

Reply:
afaics in some quick research the 64 char limit applies there as well. The issue is primarily exhibited when doing gc on extant resources provisioned by custodian that don't correspond to current policies, where re rely on this prefix to determine custodian resources, we should probably add this to a base policy validation.